plugins/available/goenv.plugin.bash

```diff
@@ -0,0 +1,18 @@
+cite about-plugin
+about-plugin 'load goenv, if you are using it'
+
+# Don't modify the environment if we can't find the tool:
+# - Check if in $PATH already
+# - Check if installed manually to $HOME
+_command_exists goenv ||
+  [[ -f "$HOME/.goenv/bin/goenv" ]] ||
+  return
+
+# Set GOENV_ROOT, if not already set
+export GOENV_ROOT="${GOENV_ROOT:-$HOME/.goenv}"
+
+# Add GOENV_ROOT/bin to PATH, if not already set
+pathmunge "$GOENV_ROOT/bin"
+
+# Initialize goenv, if it hasn't been already
+type goenv 2>/dev/null | grep -q 'is a function' || eval "$(goenv init - bash)"
```

Comment:
I would recommend against trying to be too clever in avoiding the double-initiation:
* The existence of the function is not publicly supported in the goenv documentation, so we'd be diving into the private internal parts of the tool
* Testing locally suggests its safe to double-initialize since the function delegates commands to the binary which safely re-inits the function and other variables.

What you might consider, tho, is confirming that goenv is on the path before executing the eval.

If the context of the notes in this review pass, it is possible to get here and still have not have goenv on the path.

You might prefer to fail loudly in that case, which seems reasonable, since the user asked for the plugin, maybe they should be alerted to the error - Not sure how that fits in with the usual mantra of limited error messaging.

Reply:
sure

Reply:
w/e. you already tested re-running `goenv init -` and i only wrote this check because of a complaint you made! Originally I was just running `goenv init -` _just like every other env plugin_ (notice a trend here?).